src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -107,6 +107,7 @@ object ScalafixPlugin extends AutoPlugin {
                 .map(_ => oldCompile)
             else Def.task(oldCompile)
           }.value,
+          scalafix / compile := Defaults.compileTask.value,
```

Comment:
Only [compileInputSettings are scoped within `compile` ](https://github.com/sbt/sbt/blob/0c6d7b8f300f1e405e2a2db5d84f72d30a7ec901/main/src/main/scala/sbt/Defaults.scala#L880) and all lookups of these are explicitly scoped to `compile`, so this is OK for all existing sbt versions, and hopefully all remaining 1.x.

There is a small risk that some sbt plugin customize the vanilla `compile` in such a way that our vanilla invocation fails on the build, but I am willing to take that risk and tackle it then.